source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -415,6 +422,32 @@ bool JsonTranscoderFilter::readToBuffer(Protobuf::io::ZeroCopyInputStream& strea
   return false;
 }
 
+void JsonTranscoderFilter::buildResponseFromHttpBodyOutput(Http::HeaderMap& response_headers,
+                                                           Buffer::Instance& data) {
+  try {
```

Comment:
instead of parsing JSON here, it would be better to parse protobuf directly using `Grpc::Frame` and `Grpc::Decoder`. It is cleaner and more efficient, hardcoded `"contentType"` won't work when `preserve_field_names`  is set to true.

Reply:
Cool, this is new to me. Thanks for pointing it out.